fe/fe-core/src/main/java/org/apache/doris/resource/workloadschedpolicy/WorkloadSchedPolicy.java

```diff
@@ -183,7 +183,11 @@ public void updateProperty(Map<String, String> property, List<Long> wgIdList) {
             this.priority = Integer.parseInt(priorityStr);
         }
 
-        if (wgIdList.size() > 0) {
+        String workloadGroupIdStr = property.get(WORKLOAD_GROUP);
+        // workloadGroupIdStr != null means user set workload group property,
+        // then we should overwrite policy's workloadGroupIdList
+        // if workloadGroupIdStr.length == 0, it means the policy should match all query.
+        if (workloadGroupIdStr != null) {
```

Comment:
using strings.isnullorempty to check the value is null or empty, not only check nullable.